python/lsst/imgserv/skymapStitch.py

```diff
@@ -86,29 +85,82 @@ def getSkyMap(ctrCoord, width, height, filt, source, mapType, patchType):
             patchImg = patchExposure.getMaskedImage()
             srcViewImg <<= patchImg
 
-    # Copy the pixels from the patches into the destination exposures.
+    # Copy the pixels from the source exposures to the destination exposures.
     destExposureList = []
-    for srcExpo in srcExposureList:
+    for j, srcExpo in enumerate(srcExposureList):
         sImg = srcExpo.getMaskedImage()
         srcWcs = srcExpo.getWcs()
-        dExpo = afwImage.ExposureF(destBBox, srcWcs)
+        if j == 0:
+            dBBox = destBBox # destBBox only correct for first image
+        else:
+            # Determine the correct BBox (in pixels) for the current srcWcs
+            llCorner = afwGeom.Point2I(srcWcs.skyToPixel(destCornerCoords[0]))
+            urCorner = afwGeom.Point2I(srcWcs.skyToPixel(destCornerCoords[2]))
+            # Handle negative values for dBBox llCorner.
+            if llCorner.getX() < 0:
+                llCorner.setX(0)
+            if llCorner.getY() < 0:
+                llCorner.setY(0)
+            dBBox = afwGeom.Box2I(llCorner, urCorner)
+        log.info("j={} dBBox={} sBBox={}".format(j, dBBox, srcExpo.getBBox()))
+        dExpo = afwImage.ExposureF(dBBox, srcWcs)
         dImg = dExpo.getMaskedImage()
-        beginX = destBBox.getBeginX() - sImg.getX0()
-        endX = destBBox.getEndX() - sImg.getX0()
-        beginY = destBBox.getBeginY() - sImg.getY0()
-        endY = destBBox.getEndY() - sImg.getY0()
-        dImg <<= sImg[beginX:endX, beginY:endY]
+        beginX = dBBox.getBeginX() - sImg.getX0()
+        endX = dBBox.getEndX() - sImg.getX0()
+        beginY = dBBox.getBeginY() - sImg.getY0()
+        endY = dBBox.getEndY() - sImg.getY0()
+
+        newWidth = srcExpo.getBBox().getEndX() - dBBox.getBeginX()
+        newHeight = srcExpo.getBBox().getEndY() - dBBox.getBeginY()
+        log.info("beginX={} endX={}".format(beginX, endX))
+        log.info("newWidth{} = sBBox.EndX{} - sBBox.BeginX{}".format(newWidth, srcExpo.getBBox().getEndX(), dBBox.getBeginX()))
+        log.info("beginY={} endY={}".format(beginY, endY))
+        log.info("newHeight{} = sBBox.EndY{} - sBBox.BeginY{}".format(newHeight, srcExpo.getBBox().getEndY(), dBBox.getBeginY()))
+        dImg[0:newWidth, 0:newHeight] <<= sImg[beginX:endX, beginY:endY]
```

Comment:
It's because it's the "assign pixels" operator.  In this case, "dImg[...] = sImg[...]" would work.  You used to need it to write "dImg <<= sImg"  (as "dImg = sImg" would redefine dImg), but "dImg[:] = sImg" (now) works and is to be preferred.